src/log.c

```diff
@@ -355,6 +358,14 @@ bool qd_log_enabled(qd_log_source_t *source, qd_log_level_t level) {
 
 void qd_log_impl(qd_log_source_t *source, qd_log_level_t level, const char *file, int line, const char *fmt, ...)
 {
+    /*-----------------------------------------------
+      Count this log-event in this log's histogram
+      whether or not this log is currently enabled.
+      We can always decide not to look at it later,
+      based on its used/unused status.
+    -----------------------------------------------*/
+    source->severity_histogram [ level ] ++;
```

Comment:
You have a type problem here.  qd_log_level_t can't be used as an index into an array since it is a bitmask.